src/s1reader/s1_reader.py

```diff
@@ -437,16 +438,85 @@ def burst_from_zip(zip_path: str, orbit_path: str, swath_num: int, pol: str = 'v
         raise ValueError(f"polarization not in {pols}")
 
     id_str = f'iw{swath_num}-slc-{pol}'
+
+    if not os.path.exists(path):
+        raise FileNotFoundError(f'{path} not found')
+    elif os.path.isdir(path):
+        return _burst_from_safe_dir(path, id_str, orbit_path)
+    elif os.path.isfile(path):
+        return _burst_from_zip(path, id_str, orbit_path)
+    else:
+        raise ValueError(f'{path} is unsupported')
+
+def _burst_from_zip(zip_path: str, id_str: str, orbit_path: str):
+    '''Find bursts in a Sentinel 1 zip file.
+
+    Parameters:
+    -----------
+    path : str
+        Path to zip file.
+    id_str: str
+        Identifcation of desired burst. Format: iw[swath_num]-slc-[pol]
+    orbit_path : str
+        Path the orbit file.
+
+    Returns:
+    --------
+    bursts : list
+        List of Sentinel1BurstSlc objects found in annotation XML.
+    '''
     with zipfile.ZipFile(zip_path, 'r') as z_file:
+        z_file_list = z_file.namelist()
+
         # find annotation file
-        f_annotation = [f for f in z_file.namelist() if _is_annotation_xml(f, id_str)]
+        f_annotation = [f for f in z_file_list if _is_zip_annotation_xml(f, id_str)]
         if not f_annotation:
-            raise ValueError(f"polarization {pol} not in SAFE: {zip_path}")
+            raise ValueError(f"burst {id_str} not in SAFE: {zip_path}")
         f_annotation = f_annotation[0]
 
         # find tiff file
-        f_tiff = [f for f in z_file.namelist() if 'measurement' in f and id_str in f and 'tiff' in f][0]
-        f_tiff = f'/vsizip/{zip_path}/{f_tiff}'
+        f_tiff = [f for f in z_file_list
+                  if 'measurement' in f and id_str in f and 'tiff' in f]
+        f_tiff = f'/vsizip/{zip_path}/{f_tiff[0]}' if f_tiff else ''
 
```

Comment:
if f_tiff is empty, let's let the user know. Same warning as you added at line 517.